src/ophyd_async/epics/motion/motor.py

```diff
@@ -43,13 +88,59 @@ def __init__(self, prefix: str, name="") -> None:
         self.motor_stop = epics_signal_x(prefix + ".STOP")
         # Whether set() should complete successfully or not
         self._set_success = True
+
+        self.fly_info: Optional[FlyMotorInfo] = None
+
+        # end_position with run_up_distance added on.
+        self._fly_completed_position: Optional[float] = None
+
+        self._fly_status: Optional[WatchableAsyncStatus] = None
+
         super().__init__(name=name)
 
     def set_name(self, name: str):
         super().set_name(name)
         # Readback should be named the same as its parent in read()
         self.user_readback.set_name(name)
 
+    @AsyncStatus.wrap
+    async def prepare(self, value: FlyMotorInfo):
+        """Calculate required velocity and run-up distance, then if motor limits aren't
+        breached, move to start position minus run-up distance"""
+
+        self.fly_info = value
+
+        # Velocity, at which motor travels from start_position to end_position, in motor
+        # egu/s.
+        fly_velocity = await self._prepare_velocity(
+            self.fly_info.start_position,
+            self.fly_info.end_position,
+            self.fly_info.time_for_move,
+        )
+
+        # start_position with run_up_distance added on.
+        fly_prepared_position = await self._prepare_motor_path(
+            abs(fly_velocity), self.fly_info.start_position, self.fly_info.end_position
+        )
+
+        await self.set(fly_prepared_position)
```

Comment:
Prepare shouldn't ever take very long so it might not be needed, but we could make it run in the background instead, like with `kickoff`